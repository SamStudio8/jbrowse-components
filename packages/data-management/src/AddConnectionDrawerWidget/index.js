import { ConfigurationSchema } from '@gmod/jbrowse-core/configuration'

export const ReactComponent = import('./components/AddConnectionDrawerWidget')
export { default as stateModel } from './model'
export const configSchema = ConfigurationSchema('AddConnectionDrawerWidget', {})
