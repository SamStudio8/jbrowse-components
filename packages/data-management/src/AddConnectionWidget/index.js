import { ConfigurationSchema } from '@gmod/jbrowse-core/configuration'

export const ReactComponent = import('./components/AddConnectionWidget')
export { default as stateModel } from './model'
export const configSchema = ConfigurationSchema('AddConnectionWidget', {})
