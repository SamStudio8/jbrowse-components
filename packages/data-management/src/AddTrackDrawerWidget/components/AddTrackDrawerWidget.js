import { getSession } from '@gmod/jbrowse-core/util'
import Button from '@material-ui/core/Button'
import Step from '@material-ui/core/Step'
import StepContent from '@material-ui/core/StepContent'
import StepLabel from '@material-ui/core/StepLabel'
import Stepper from '@material-ui/core/Stepper'
import { withStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import { observer, PropTypes as MobxPropTypes } from 'mobx-react'
import propTypes from 'prop-types'
import React, { useState } from 'react'
import { readConfObject } from '@gmod/jbrowse-core/configuration'
import ConfirmTrack from './ConfirmTrack'
import TrackSourceSelect from './TrackSourceSelect'

const styles = theme => ({
  root: {
    marginTop: theme.spacing(1),
  },
  stepper: {
    backgroundColor: theme.palette.background.default,
  },
  button: {
    marginTop: theme.spacing(1),
    marginRight: theme.spacing(1),
  },
  actionsContainer: {
    marginBottom: theme.spacing(2),
  },
  stepContent: {
    margin: theme.spacing(1),
  },
})

const steps = ['Enter track data', 'Confirm track type']

function AddTrackDrawerWidget(props) {
  const [activeStep, setActiveStep] = useState(0)
  const [trackSource, setTrackSource] = useState('fromFile')
  const [trackData, setTrackData] = useState({ uri: '' })
  const [trackName, setTrackName] = useState('')
  const [trackType, setTrackType] = useState('')
  const [trackAdapter, setTrackAdapter] = useState({})
  const [datasetName, setDatasetName] = useState('')

  const { classes, model } = props

  const session = getSession(model)

  function getStepContent() {
    switch (activeStep) {
      case 0:
        return (
          <TrackSourceSelect
            trackSource={trackSource}
            setTrackSource={setTrackSource}
            trackData={trackData}
            setTrackData={setTrackData}
          />
        )
      case 1:
        return (
          <ConfirmTrack
            session={session}
            trackData={trackData}
            trackName={trackName}
            setTrackName={setTrackName}
            trackType={trackType}
            setTrackType={setTrackType}
            trackAdapter={trackAdapter}
            setTrackAdapter={setTrackAdapter}
            datasetName={datasetName}
            setDatasetName={setDatasetName}
          />
        )
      default:
        return <Typography>Unknown step</Typography>
    }
  }

  function handleNext() {
    if (activeStep !== steps.length - 1) {
      setActiveStep(activeStep + 1)
      return
    }
    trackAdapter.features = trackData.config
    const trackConf = session.datasets
      .find(dataset => readConfObject(dataset, 'name') === datasetName)
      .addTrackConf({
        type: trackType,
        name: trackName,
        adapter: trackAdapter,
      })
    model.view.showTrack(trackConf)
    session.hideDrawerWidget(model)
  }

  function handleBack() {
    setActiveStep(activeStep - 1)
  }

  function isNextDisabled() {
    switch (activeStep) {
      case 0:
        return !(trackData.uri || trackData.localPath || trackData.config)
      case 1:
        return !(trackName && trackType && trackAdapter.type && datasetName)
      default:
        return true
    }
  }
  console.log(steps)

  return (
    <div className={classes.root}>
      <Stepper
        className={classes.stepper}
        activeStep={activeStep}
        orientation="vertical"
      >
        {steps.map(label => (
          <Step key={label}>
            <StepLabel>{label}</StepLabel>
            <StepContent>
              {getStepContent()}
              <div className={classes.actionsContainer}>
                <Button
                  disabled={activeStep === 0}
                  onClick={handleBack}
                  className={classes.button}
                >
                  Back
                </Button>
                <Button
                  disabled={isNextDisabled()}
                  variant="contained"
                  color="primary"
                  onClick={handleNext}
                  className={classes.button}
                  data-testid="addTrackNextButton"
                >
                  {activeStep === steps.length - 1 ? 'Add' : 'Next'}
                </Button>
              </div>
            </StepContent>
          </Step>
        ))}
      </Stepper>
    </div>
  )
}

AddTrackDrawerWidget.propTypes = {
  classes: propTypes.objectOf(propTypes.string).isRequired,
  model: MobxPropTypes.observableObject.isRequired,
}

export default withStyles(styles)(observer(AddTrackDrawerWidget))
