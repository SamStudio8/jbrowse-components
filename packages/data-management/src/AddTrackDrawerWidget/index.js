import { ConfigurationSchema } from '@gmod/jbrowse-core/configuration'

export const ReactComponent = import('./components/AddTrackDrawerWidget')
export { default as stateModelFactory } from './model'
export const configSchema = ConfigurationSchema('AddTrackDrawerWidget', {})
