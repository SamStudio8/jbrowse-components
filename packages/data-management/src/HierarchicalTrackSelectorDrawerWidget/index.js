import { ConfigurationSchema } from '@gmod/jbrowse-core/configuration'

export const ReactComponent = import('./components/HierarchicalTrackSelector')
export { default as stateModelFactory } from './model'
export const configSchema = ConfigurationSchema(
  'HierarchicalTrackSelectorDrawerWidget',
  {},
)
