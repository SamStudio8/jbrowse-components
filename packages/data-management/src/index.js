import { lazy } from 'react'
import ConnectionType from '@gmod/jbrowse-core/pluggableElementTypes/ConnectionType'
import DrawerWidgetType from '@gmod/jbrowse-core/pluggableElementTypes/DrawerWidgetType'
import Plugin from '@gmod/jbrowse-core/Plugin'
import {
  configSchema as ucscConfigSchema,
  modelFactory as ucscModelFactory,
} from './ucsc-trackhub'
import {
  configSchema as jbrowse1ConfigSchema,
  modelFactory as jbrowse1ModelFactory,
} from './jbrowse1'
import {
  reactComponent as AddTrackReactComponent,
  stateModelFactory as AddTrackStateModelFactory,
  configSchema as AddTrackConfigSchema,
} from './AddTrackDrawerWidget'
import {
  reactComponent as AddConnectionReactComponent,
  stateModel as AddConnectionStateModel,
  configSchema as AddConnectionConfigSchema,
} from './AddConnectionDrawerWidget'
import {
  reactComponent as HierarchicalTrackSelectorReactComponent,
  stateModelFactory as HierarchicalTrackSelectorStateModelFactory,
  configSchema as HierarchicalTrackSelectorConfigSchema,
} from './HierarchicalTrackSelectorDrawerWidget'

export default class extends Plugin {
  install(pluginManager) {
    pluginManager.addConnectionType(
      () =>
        new ConnectionType({
          name: 'UCSCTrackHubConnection',
          configSchema: ucscConfigSchema,
          stateModel: ucscModelFactory(pluginManager),
          displayName: 'UCSC Track Hub',
          description: 'A track or assembly hub in the Track Hub format',
          url: '//genome.ucsc.edu/goldenPath/help/hgTrackHubHelp.html#Intro',
        }),
    )

    pluginManager.addConnectionType(
      () =>
        new ConnectionType({
          name: 'JBrowse1Connection',
          configSchema: jbrowse1ConfigSchema,
          stateModel: jbrowse1ModelFactory(pluginManager),
          displayName: 'JBrowse 1 Data',
          description: 'A JBrowse 1 data directory',
          url: '//jbrowse.org/',
        }),
    )

    pluginManager.addDrawerWidgetType(() => {
      return new DrawerWidgetType({
        name: 'HierarchicalTrackSelectorDrawerWidget',
        heading: 'Available Tracks',
        configSchema: HierarchicalTrackSelectorConfigSchema,
        stateModel: HierarchicalTrackSelectorStateModelFactory(pluginManager),
        LazyReactComponent: lazy(() => HierarchicalTrackSelectorReactComponent),
      })
    })

    pluginManager.addDrawerWidgetType(() => {
      return new DrawerWidgetType({
        name: 'AddTrackDrawerWidget',
        heading: 'Add a track',
        configSchema: AddTrackConfigSchema,
        stateModel: AddTrackStateModelFactory(pluginManager),
        LazyReactComponent: lazy(() => AddTrackReactComponent),
      })
    })

    pluginManager.addDrawerWidgetType(() => {
      return new DrawerWidgetType({
        name: 'AddConnectionDrawerWidget',
        heading: 'Add a connection',
        configSchema: AddConnectionConfigSchema,
        stateModel: AddConnectionStateModel,
        LazyReactComponent: lazy(() => AddConnectionReactComponent),
      })
    })
  }
}
