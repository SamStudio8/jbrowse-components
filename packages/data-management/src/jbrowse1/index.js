import { ConfigurationSchema } from '@gmod/jbrowse-core/configuration'

export { default as modelFactory } from './model'

export const configSchema = ConfigurationSchema(
  'JBrowse1Connection',
  {
    name: {
      type: 'string',
      defaultValue: 'nameOfJBrowse1Connection',
    },
    dataDirLocation: {
      type: 'fileLocation',
      defaultValue: { uri: '/path/to/data/dir/' },
    },
    assemblyName: {
      type: 'string',
      defaultValue: '',
    },
    useAssemblySequences: {
      type: 'boolean',
      defaultValue: false,
    },
  },
  { explicitlyTyped: true },
)
