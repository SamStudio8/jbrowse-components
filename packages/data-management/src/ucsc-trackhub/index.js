import { ConfigurationSchema } from '@gmod/jbrowse-core/configuration'

export { default as modelFactory } from './model'

export const configSchema = ConfigurationSchema(
  'UCSCTrackHubConnection',
  {
    name: {
      type: 'string',
      defaultValue: 'nameOfUCSCTrackHubConnection',
      description: 'a unique name for this connection',
    },
    hubTxtLocation: {
      type: 'fileLocation',
      defaultValue: { uri: 'http://mysite.com/path/to/hub.txt' },
      description: 'location of the hub file (usually called hub.txt)',
    },
    assemblyNames: {
      type: 'stringArray',
      defaultValue: [],
      description:
        'names of the assemblies (genomes) in this hub to connect. if left empty, all assemblies will be connected',
    },
    useAssemblySequences: {
      type: 'stringArray',
      defaultValue: [],
      description:
        'name of any assemblies to use to override an existing assembly sequence. probably safe to ignore this.',
    },
  },
  { explicitlyTyped: true },
)
