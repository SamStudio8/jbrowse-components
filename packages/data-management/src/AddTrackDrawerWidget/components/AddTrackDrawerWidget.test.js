import React from 'react'
import {
  render,
  cleanup,
  fireEvent,
  waitForElement,
} from '@testing-library/react'
import { createTestSession } from '@gmod/jbrowse-web/src/rootModel'
import AddTrackDrawerWidget from './AddTrackDrawerWidget'

jest.mock('@material-ui/core/Select', () => ({ children, onChange }) => {
  return (
    <select data-testid="select" onChange={onChange}>
      {children}
    </select>
  )
})

jest.mock('@material-ui/core/MenuItem', () => ({ value, children }) => {
  return <option value={value}>{children}</option>
})

describe('<AddTrackDrawerWidget />', () => {
  let session
  let model

  beforeAll(() => {
    session = createTestSession()
    session.addDataset({
      name: 'volvox',
      assembly: {
        name: 'volMyt1',
        sequence: {
          type: 'ReferenceSequenceTrack',
          adapter: {
            type: 'FromConfigAdapter',
            features: [
              {
                refName: 'ctgA',
                uniqueId: 'firstId',
                start: 0,
                end: 1000,
                seq:
                  'cattgttgcggagttgaacaACGGCATTAGGAACACTTCCGTCTCtcacttttatacgattatgattggttctttagccttggtttagattggtagtagtagcggcgctaatgctacctgaattgagaactcgagcgggggctaggcaaattctgattcagcctgacttctcttggaaccctgcccataaatcaaagggttagtgcggccaaaacgttggacaacggtattagaagaccaacctgaccaccaaaccgtcaattaaccggtatcttctcggaaacggcggttctctcctagatagcgatctgtggtctcaccatgcaatttaaacaggtgagtaaagattgctacaaatacgagactagctgtcaccagatgctgttcatctgttggctccttggtcgctccgttgtacccaggctactttgaaagagcgcagaatacttagacggtatcgatcatggtagcatagcattctgataacatgtatggagttcgaacatccgtctggggccggacggtccgtttgaggttggttgatctgggtgatagtcagcaagatagacgttagataacaaattaaaggattttaccttagattgcgactagtacaacggtacatcggtgattcgcgctctactagatcacgctatgggtaccataaacaaacggtggaccttctcaagctggttgacgcctcagcaacataggcttcctcctccacgcatctcagcataaaaggcttataaactgcttctttgtgccagagcaactcaattaagcccttggtaccgtgggcacgcattctgtcacggtgaccaactgttcatcctgaatcgccgaatgggactatttggtacaggaatcaagcggatggcactactgcagcttatttacgacggtattcttaaagtttttaagacaatgtatttcatgggtagttcggtttgttttattgctacacaggctcttgtagacgacctacttagcactacgg',
              },
            ],
          },
        },
      },
      tracks: [
        {
          configId: 'i3jUPmrgMOS',
          type: 'FilteringTrack',
          name: 'Filter Test',
          adapter: {
            type: 'FromConfigAdapter',
            features: [
              {
                uniqueId: 'one',
                refName: 'ctgA',
                start: 100,
                end: 101,
                type: 'foo',
                name: 'Boris',
                note: 'note for boris',
              },
              {
                uniqueId: 'two',
                refName: 'ctgA',
                start: 110,
                end: 111,
                type: 'bar',
                name: 'Theresa',
                note: 'note for theresa',
              },
              {
                uniqueId: 'three',
                refName: 'ctgA',
                start: 120,
                end: 121,
                type: 'baz',
                name: 'Nigel',
                note: 'note for nigel',
              },
              {
                uniqueId: 'four',
                refName: 'ctgA',
                start: 130,
                end: 131,
                type: 'quux',
                name: 'Geoffray',
                note: 'note for geoffray',
              },
            ],
          },
          renderer: {
            type: 'SvgFeatureRenderer',
            labels: {},
          },
          filterAttributes: ['type', 'start', 'end'],
        },
      ],
    })
    const view = session.addLinearGenomeViewOfDataset('volvox')
    model = session.addDrawerWidget(
      'AddTrackDrawerWidget',
      'addTrackDrawerWidget',
      { view: view.id },
    )
  })

  afterEach(cleanup)

  it('renders', () => {
    const { container } = render(<AddTrackDrawerWidget model={model} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('adds a track', async () => {
    const {
      container,
      debug,
      getByTestId,
      getAllByTestId,
      getByText,
      getAllByRole,
    } = render(<AddTrackDrawerWidget model={model} />)
    expect(session.datasets[0].tracks.length).toBe(1)
    fireEvent.click(getByTestId('addTrackFromConfigRadio'))
    fireEvent.click(getByTestId('addTrackNextButton'))
    console.log(getAllByTestId('trackNameInput'))
    fireEvent.change(getAllByTestId('trackNameInput')[1], {
      target: { value: 'Test track name' },
    })
    // fireEvent.click(getAllByRole('button')[0])
    expect(container).toMatchSnapshot()
    const e = await waitForElement(() => getAllByTestId('select'))
    fireEvent.change(e[2], {
      target: { value: 'AlignmentsTrack' },
    })
    fireEvent.change(e[3], {
      target: { value: 'volvox' },
    })
    expect(session.datasets[0].tracks.length).toBe(1)

    console.log(getAllByTestId('addTrackNextButton').length)

    fireEvent.click(getAllByTestId('addTrackNextButton')[0])

    expect(session.datasets[0].tracks.length).toBe(2)
  })
})
