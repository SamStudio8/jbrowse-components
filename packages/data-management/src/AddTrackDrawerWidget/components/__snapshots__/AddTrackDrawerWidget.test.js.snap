// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTrackDrawerWidget /> renders 1`] = `
<div
  class="AddTrackDrawerWidget-root-1"
>
  <div
    class="MuiPaper-root MuiPaper-elevation0 MuiStepper-root MuiStepper-vertical AddTrackDrawerWidget-stepper-2"
  >
    <div
      class="MuiStep-root MuiStep-vertical"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-vertical"
      >
        <span
          class="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              class="MuiStepIcon-text"
              text-anchor="middle"
              x="12"
              y="16"
            >
              1
            </text>
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
          >
            Enter track data
          </span>
        </span>
      </span>
      <div
        class="MuiStepContent-root"
        icon="1"
      >
        <div
          class="MuiCollapse-container MuiStepContent-transition MuiCollapse-entered"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                class="TrackSourceSelect-root-105"
              >
                <fieldset
                  class="MuiFormControl-root"
                >
                  <div
                    aria-label="Data location"
                    class="MuiFormGroup-root"
                    role="radiogroup"
                  >
                    <label
                      class="MuiFormControlLabel-root"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-124 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-125 Mui-checked MuiIconButton-colorSecondary"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <div
                            class="PrivateRadioButtonIcon-root-140 PrivateRadioButtonIcon-checked-142"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              role="presentation"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                              />
                            </svg>
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-141"
                              focusable="false"
                              role="presentation"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                              />
                            </svg>
                          </div>
                          <input
                            checked=""
                            class="PrivateSwitchBase-input-127"
                            data-testid="addTrackFromFileRadio"
                            type="radio"
                            value="fromFile"
                          />
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                      >
                        From file
                      </span>
                    </label>
                    <label
                      class="MuiFormControlLabel-root"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-124 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <div
                            class="PrivateRadioButtonIcon-root-140"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              role="presentation"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                              />
                            </svg>
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-141"
                              focusable="false"
                              role="presentation"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                              />
                            </svg>
                          </div>
                          <input
                            class="PrivateSwitchBase-input-127"
                            data-testid="addTrackFromConfigRadio"
                            type="radio"
                            value="fromConfig"
                          />
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                      >
                        From configuration
                      </span>
                    </label>
                  </div>
                </fieldset>
                <div
                  class="MuiPaper-root MuiPaper-elevation1 TrackSourceSelect-paper-106 MuiPaper-rounded"
                >
                  <fieldset
                    class="MuiFormControl-root"
                  >
                    <div
                      class="MuiFormGroup-root"
                      role="radiogroup"
                    >
                      <label
                        class="MuiFormControlLabel-root"
                      >
                        <span
                          aria-disabled="false"
                          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-124 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-125 Mui-checked MuiIconButton-colorSecondary"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <div
                              class="PrivateRadioButtonIcon-root-140 PrivateRadioButtonIcon-checked-142"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root"
                                focusable="false"
                                role="presentation"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                                />
                              </svg>
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-141"
                                focusable="false"
                                role="presentation"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                                />
                              </svg>
                            </div>
                            <input
                              checked=""
                              class="PrivateSwitchBase-input-127"
                              type="radio"
                              value="uri"
                            />
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </span>
                        <span
                          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                        >
                          URL
                        </span>
                      </label>
                      <label
                        class="MuiFormControlLabel-root Mui-disabled"
                      >
                        <span
                          aria-disabled="true"
                          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-124 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-disabled-126 Mui-disabled MuiIconButton-colorSecondary Mui-disabled Mui-disabled"
                          tabindex="-1"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <div
                              class="PrivateRadioButtonIcon-root-140"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root"
                                focusable="false"
                                role="presentation"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                                />
                              </svg>
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-141"
                                focusable="false"
                                role="presentation"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                                />
                              </svg>
                            </div>
                            <input
                              class="PrivateSwitchBase-input-127"
                              disabled=""
                              type="radio"
                              value="localPath"
                            />
                          </span>
                        </span>
                        <span
                          class="MuiTypography-root MuiFormControlLabel-label Mui-disabled MuiTypography-body1"
                        >
                          Local File
                        </span>
                      </label>
                    </div>
                  </fieldset>
                  <div
                    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                      data-shrink="false"
                    >
                      fileLocation
                    </label>
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input"
                        type="text"
                        value=""
                      />
                    </div>
                    
                  </div>
                </div>
              </div>
              <div
                class="AddTrackDrawerWidget-actionsContainer-4"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root AddTrackDrawerWidget-button-3 MuiButton-text Mui-disabled Mui-disabled"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Back
                  </span>
                </button>
                <button
                  class="MuiButtonBase-root MuiButton-root AddTrackDrawerWidget-button-3 MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
                  data-testid="addTrackNextButton"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Next
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiStepConnector-root MuiStepConnector-vertical Mui-disabled"
    >
      <span
        class="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div
      class="MuiStep-root MuiStep-vertical"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-vertical Mui-disabled"
      >
        <span
          class="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              class="MuiStepIcon-text"
              text-anchor="middle"
              x="12"
              y="16"
            >
              2
            </text>
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
          >
            Confirm track type
          </span>
        </span>
      </span>
      <div
        class="MuiStepContent-root MuiStepContent-last"
        disabled=""
        icon="2"
      />
    </div>
  </div>
</div>
`;
